### PR TITLE
Enhance Calendar conversation

### PR DESCRIPTION
**Pull request by Hobbitsloth**
enhancement of the speech input

Pull request status:
- [x] Tests added/updated
- [x] 80 % Coverage
- [x] Code conventions
- [x] [Doku required and added](https://github.com/AmyAssist/Amy/wiki/Calendar-Plugin)
- [x] Ready for review

### CODE REVIEW

File: plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeech.java [hunk 1]
```diff
@@ -125,18 +98,23 @@ public String getEventsAt(Map<String, EntityData> entities) {
...
 			chosenDate = chosenDate.withYear(this.environment.getCurrentLocalDateTime().getYear() + 1);
 		}
 		events = this.logic.getEventsAt(chosenDate);
-		if (events.isEmpty()) {
-			return "No events found for the " + getDate(chosenDate) + " " + chosenDate.getYear() + ".";
-		}
 		for (CalendarEvent event : events) {
 			eventList.add(this.checkDate(chosenDate, event, false));
 		}
+		if (entities.get("date").getString().equals("today") || entities.get("date").getString().equals("tomorrow")) {
```
Comment (FlxB2) [inline plugins/calendar/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeech.java]: create a String for entities.get("date") and then compare with the string. 
You should also use it later to insert it into your return value

File: plugins/calendar/src/main/resources/CalendarSpeech.aim.xml [hunk 2]
```diff
@@ -96,35 +83,35 @@
...
 			<gram>[at] [the] {start}</gram>
 		</Prompt>		
 		<Prompt entityTemplateId="end">
-			<text>When does the event end?</text>
+			<text>When does it end?</text>
 			<gram>[at] [the] {end}</gram>
 		</Prompt>
 		<Prompt entityTemplateId="description">
-			<text>You can add a description to the event or just type no if you don't want to.</text>
+			<text>You can skip each of the following two questions by saying "no" after I asked them. What should I write in the description?</text>
```
Comment (FlxB2) [inline plugins/calendar/src/main/resources/CalendarSpeech.aim.xml]: Be careful with this. The order of prompts amy asks might be changed in the future. 
For now it's all right. I might have to add a specific order attribute to the .aim.xml specification..

File: plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeechTest.java [hunk 3]
```diff
@@ -114,45 +112,94 @@ private void mockEntityData() {
...
-		verify(this.logic).getEvents(1);
+		String response = this.speech.getEvents(map);
+		assertThat(response, equalToIgnoringWhiteSpace("No events found."));
+	}
+	
+	/**
+	 * test getEvents()
+	 */
+	@Test
+	void testGetEvents2() {
```
Comment (FlxB2) [inline plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeechTest.java]: better description, and method name. If this test fails nobody will know why

File: plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeechTest.java [hunk 4]
```diff
@@ -114,45 +112,94 @@ private void mockEntityData() {
...
+		when(this.environment.getCurrentLocalDateTime()).thenReturn(LocalDateTime.of(2018, 9, 20, 10, 00));
+		String response = this.speech.getEvents(map);
+		assertThat(response, equalToIgnoringWhiteSpace("You have following upcoming event: Summary from 10:00 am until 12:00 pm."));
+	}
+	
+	/**
+	 * test getEvents()
+	 */
+	@Test
+	void testGetEvents3() {
```
Comment (FlxB2) [inline plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeechTest.java]: same as above

File: plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeechTest.java [hunk 5]
```diff
@@ -114,45 +112,94 @@ private void mockEntityData() {
...
-		verify(this.logic).getEventsAt(LocalDateTime.of(2018, 9, 20, 10, 00));	
+		String response = this.speech.getEventsAt(map);
+		assertThat(response, equalToIgnoringWhiteSpace("You have no events today."));
+	}
+	
+	/**
+	 * test getEventsAt() with today and events
+	 */
+	@Test
+	void testGetEventsToday2() {
```
Comment (FlxB2) [inline plugins/calendar/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/calendar/CalendarSpeechTest.java]: same as above

File: plugins/calendar/src/main/resources/CalendarSpeech.aim.xml (hunk 2 above)
Comment (Hobbitsloth) [reply plugins/calendar/src/main/resources/CalendarSpeech.aim.xml]: Yes I know and considered taking the risk for now

**Pull request merged**